source/adios2/core/ADIOS.cpp

```diff
@@ -79,6 +79,18 @@ IO &ADIOS::DeclareIO(const std::string ioName)
     return ioPair.first->second;
 }
 
+IO &ADIOS::GetIO(const std::string name)
+{
+    auto itIO = m_IOs.find(name);
+    if (itIO == m_IOs.end())
```

Comment:
> the user has the choice to turn off extra checks, not us.

Sure, but some check's aren't extras.  In this case, what should the behavior be if the requested object is not found, i.e. what get's returned.